app/uk/gov/hmrc/agentsubscriptionfrontend/service/SubscriptionService.scala

```diff
@@ -61,4 +82,50 @@ class SubscriptionService @Inject()(agentSubscriptionConnector: AgentSubscriptio
       case e => throw e
     }
   }
+
+  def completePartialSubscription(details: CompletePartialSubscriptionBody)(
+    implicit hc: HeaderCarrier,
+    ec: ExecutionContext): Future[(Boolean, Option[Arn])] =
+    for {
+      subscriptionProgress <- getSubscriptionStatus(details.utr, details.knownFacts.postcode) //FixMe APB-2805 this check should happen previously and can be stored in the session
+      isPartiallySubscribed = subscriptionProgress.state == SubscriptionState.IsOnlySubscribedInETMP
+      (wasPartiallySubscribed, optSucceededAllocatingArn) <- isPartiallySubscribed match {
+                                                              case true =>
+                                                                agentSubscriptionConnector
+                                                                  .completePartialSubscription(details)
+                                                                  .map(arn => (true, Some.apply(arn)))
+                                                                  .recover {
+                                                                    case e: Upstream4xxResponse
+                                                                        if Seq(Status.FORBIDDEN, Status.CONFLICT) contains e.upstreamResponseCode =>
+                                                                      (true, None)
+                                                                    case e => throw e
+                                                                  }
+                                                              case false =>
```

Comment:
Looks like this case lead to showAlreadySubscribed in CheckAgencyController since we can get false from any SubscriptionState other than IsOnlySubscribedInETMP

Reply:
true or false does not change much actually, you need Some(arn) to be successful.